src/g_game.c

```diff
@@ -2802,7 +2802,7 @@ static int G_GetHelpers(void)
     //
 
     j = M_CheckParm ("-dogs");
-  return j ? j+1 < myargc ? M_ParmArgToInt(j) : 1 : default_dogs;
+  return j ? (j+1 < myargc && myargv[j+1][0] != '-') ? M_ParmArgToInt(j) : 1 : default_dogs;
```

Comment:
If we replace `default_dogs` with `(demo_version < 203 ? 0 : default_dogs)` here, we can check for `dogs` in the condition below (L 3116) and don't have to check for both command line parameters again.

Reply:
But `demo_version` is not set when calling `G_GetHelpers()`.

Reply:
Ah, right!